File: Efficient_sorting_algorithms/Counting_sort/counting_sort.js
function countingSort(arr, max) {
    let countingArray = new Array(max + 1).fill(0);
    let finalArray = [];

    for (let i = 0; i < arr.length; i++) {
        let num = arr[i];
        countingArray[num]++
    }

    for (let j = 0; j < arr.length + 1; j++) {
        let counter = countingArray[j];
        while (counter > 0) {
            finalArray.push(j);
            counter--;
        }
    }
    return finalArray
}

console.log(countingSort([4, 3, 5, 6, 7, 8, 3, 1], 8))

module.exports = {
    countingSort
};

File: Efficient_sorting_algorithms/Counting_sort/counting_sort.rb
def counting_sort(arr, max)
    counter_arr = Array.new(max + 1).fill(0)
    final_arr = []

    i = 0
    until i == arr.length
        num = arr[i]
        counter_arr[num] += 1
        i += 1
    end

    j = 0
    until j == arr.length + 1
        counter = counter_arr[j]
        while counter > 0
            final_arr << j
            counter -= 1
        end
        j += 1
    end
    final_arr
end


p counting_sort([4,3,5,6,7,8,3,1], 8)